pandas/tests/extension/test_numpy.py

```diff
@@ -0,0 +1,212 @@
+import numpy as np
+import pytest
+
+import pandas as pd
+from pandas import compat
+from pandas.core.arrays.numpy_ import PandasArray, PandasDtype
+import pandas.util.testing as tm
+
+from . import base
+
+
+@pytest.fixture
```

Comment:
you need to test for all numpy dtypes i think

Reply:
That would create ~7,000 tests. Probably not what we want.

It also won't work for types like int, since ops casting to float will mean some of the base tests expected arrays will be incorrect. That would require special casing which test methods are skipped based on the dtype.

Reply:
ok, the problem is this doesn't test for int (as you have noted), nor str, not to mention things like datetime. So either we should just restrict this, or test it.

Reply:
Note that the very basics are tested for every dtype in https://github.com/pandas-dev/pandas/pull/24227/files#diff-941bc2d6a7667d26acf010e1072c134bR1199.

Do you have suggestions here? Exploding the number of tests is a no-go I think, so can we be more targeted?

Scanning through PandasArray it looks like the things that are dtype-dependent are

* `__setitem__`

and

* PandasDtype._is_numeric
* PandasDtype._is_boolean

Do you see any others?

Reply:
I think adding just construction tests would be enough for now